label_studio/projects/functions/next_task.py

```diff
@@ -141,9 +143,30 @@ def get_not_solved_tasks_qs(user, project, prepared_tasks, assigned_flag, queue_
         user_postponed_tasks = postponed_drafts.distinct().values_list('task__pk', flat=True)
         not_solved_tasks = not_solved_tasks.exclude(pk__in=user_postponed_tasks)
 
-    # if annotator is assigned for tasks, he must to solve it regardless of is_labeled=True
+    # if annotator is assigned for tasks, he must solve it regardless of is_labeled=True
     if not assigned_flag:
-        not_solved_tasks = not_solved_tasks.filter(is_labeled=False)
+        # include tasks that have been completed if their agreement is not at threshold if threshold setting is set
+        lse_project = getattr(project, 'lse_project', None)
+        if (
+            lse_project
+            and lse_project.agreement_threshold is not None
+            and get_tasks_agreement_queryset
+            and user.is_annotator
+        ):
+            not_solved_tasks = (
+                get_tasks_agreement_queryset(not_solved_tasks)
+                # include tasks that are not labeled or are labeled but fall below the agreement threshold
+                .filter(
+                    Q(_agreement__lt=lse_project.agreement_threshold, is_labeled=True) | Q(is_labeled=False)
+                ).annotate(annotators=Count('annotations__completed_by', distinct=True))
+                # skip tasks that have been annotated by the maximum number of annotators
+                .filter(annotators__lt=lse_project.max_assignable_annotators)
+            )
+            prioritized_on_agreement, not_solved_tasks = _prioritize_low_agreement_tasks(not_solved_tasks, lse_project)
```

Comment:
Is `prioritized_on_agreement` used anywhere?

Reply:
It was a some point, I was mulling over removing that yesterday and had forgotten to remove it.  Let me have another look

Reply:
Right, so I had shelved the code that used that field.  Added it back in below in this block:
```
        if project.show_overlap_first and not prioritized_on_agreement:
            # don't output anything - just filter tasks with overlap
            logger.debug(f'User={user} tries overlap first from prepared tasks')
            _, not_solved_tasks = _try_tasks_with_overlap(not_solved_tasks)
            queue_info += 'Show overlap first'
```
This warrants some explanation because I know that block is confusing.  So normally, if show_overlap_first is set, that block of code prioritizes tasks with more than one annotation on them already (the reasoning I believe is to prioritize the tasks that are near to completion).  However, tasks with prioritized_on_agreement set to True are even closer to completion as they have already reached the completeness threshold (is_labeled is true) and they just need improved agreement scores to be completed, so in that case we don't want to reprioritize on the overlap.